src/Features/Core/Portable/AddParameter/AbstractAddParameterCodeFixProvider.cs

```diff
@@ -113,125 +137,282 @@ internal abstract class AbstractAddParameterCodeFixProvider<
+                    }
+
+                    var existingParameters = generator.GetParameters(methodNode);
+                    var insertionIndex = isNamedArgument
+                        ? existingParameters.Count
+                        : argumentList.IndexOf(argument);
+
+                    if (method.MethodKind == MethodKind.ReducedExtension)
+                    {
+                        insertionIndex++;
+                    }
+
+                    AddParameter(
+                        syntaxFacts, editor, methodNode, argument,
+                        insertionIndex, parameterDeclaration, cancellationToken);
 
-            var newRoot = editor.GetChangedRoot();
-            var newDocument = methodDocument.WithSyntaxRoot(newRoot);
+                }
+                var newRoot = editor.GetChangedRoot();
+                solution = solution.WithDocumentSyntaxRoot(document.Id, newRoot);
+            }
 
-            return newDocument;
+            return solution;
         }
 
-        private async Task<(IParameterSymbol, bool isNamedArgument)> CreateParameterSymbolAsync(
-            Document invocationDocument,
-            IMethodSymbol method,
-            TArgumentSyntax argument,
-            CancellationToken cancellationToken)
+        private static async Task<ITypeSymbol> GetArgumentTypeAsync(Document invocationDocument, TArgumentSyntax argument, CancellationToken cancellationToken)
         {
             var syntaxFacts = invocationDocument.GetLanguageService<ISyntaxFactsService>();
-            var semanticFacts = invocationDocument.GetLanguageService<ISemanticFactsService>();
-            var argumentName = syntaxFacts.GetNameForArgument(argument);
-            var expression = syntaxFacts.GetExpressionOfArgument(argument);
-
             var semanticModel = await invocationDocument.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-            var parameterType = semanticModel.GetTypeInfo(expression).Type ?? semanticModel.Compilation.ObjectType;
+            var argumentExpression = syntaxFacts.GetExpressionOfArgument(argument);
+            var argumentType = semanticModel.GetTypeInfo(argumentExpression).Type ?? semanticModel.Compilation.ObjectType;
+            return argumentType;
+        }
+
+        private static async Task<ImmutableArray<IMethodSymbol>> FindMethodDeclarationReferences(
+            Document invocationDocument, IMethodSymbol method, CancellationToken cancellationToken)
+        {
+            var progress = new StreamingProgressCollector(StreamingFindReferencesProgress.Instance);
+
+            await SymbolFinder.FindReferencesAsync(
+                symbolAndProjectId: SymbolAndProjectId.Create(method, invocationDocument.Project.Id),
+                solution: invocationDocument.Project.Solution,
+                documents: null,
+                progress: progress,
+                cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
note to @jnm2 this would be a good place for *another* FAR option.  Namely one that says "i only want hte cascading between definition.  I don't actually want the references".  That would help speed this up a lot.